### PR TITLE
Add DCO requirements and updated code of conduct

### PR DESCRIPTION
**Pull request by willtsai**
Issue reference: https://github.com/dapr/docs/issues/2039

# Description

Add DCO requirements and updated code of conduct. Content inspired and borrowed from https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work

## Issue reference

https://github.com/dapr/docs/issues/2039

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [x] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [x] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [x] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by willtsai** [comment]
we're also a bit inconsistent in that some repos point to docs for contribution.md; wonder if that's actually a better approach to take here, e.g. https://github.com/dapr/docs/blob/v1.5/CONTRIBUTING.md

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -64,20 +58,61 @@ All contributions come through pull requests. To submit a proposed change, we re
...
+#### I didn't sign my commit, now what?!
+
+No worries - You can easily replay your changes, sign them and force push them!
+
+```
+git checkout <branch-name>
+git reset $(git merge-base main <branch-name>)
+git add -A
+git commit -s -m "one commit on <branch-name>"
+git push --force
```
Comment (halspang) [inline CONTRIBUTING.md:109]: You can make this simpler by just doing an amend commit with the sign-off:

`git commit --am -s`

Saves a few commands and removes the need for a reset. Even in copy-paste scenarios, git commands can get wild so I like to keep it simpler.

**Review by halspang — COMMENTED** [review]
Overall seems fine. Proposed a simplification in the git if you want, but if you like the current setup I'm fine either way.

File: CONTRIBUTING.md (hunk 1 above)
Comment (willtsai) [reply CONTRIBUTING.md]: Thanks @halspang! I tend to agree with you here, so I've updated the commands using what DCO-bot recommends.

**Review by halspang — APPROVED** [review]

**Review by halspang — APPROVED** [review]

**Review by artursouza — APPROVED** [review]

**Pull request merged**